crates/rome_service/src/file_handlers/javascript.rs

```diff
@@ -126,18 +127,44 @@ fn debug_print(_rome_path: &RomePath, parse: AnyParse) -> String {
     format!("{tree:#?}")
 }
 
-fn lint(rome_path: &RomePath, parse: AnyParse, filter: AnalysisFilter) -> Vec<Diagnostic> {
+fn lint(
+    rome_path: &RomePath,
+    parse: AnyParse,
+    filter: AnalysisFilter,
+    rules: Option<&Rules>,
+) -> Vec<Diagnostic> {
     let tree = parse.tree();
     let mut diagnostics = parse.into_diagnostics();
 
     let file_id = rome_path.file_id();
     analyze(file_id, &tree, filter, |signal| {
-        if let Some(mut diag) = signal.diagnostic() {
+        if let Some(diagnostic) = signal.diagnostic() {
+            // We do now check if the severity of the diagnostics should be changed.
+            // The configuration allows to change the severity of the diagnostics emitted by rules.
+            let severity = if let Some(rules) = rules {
+                if let Some(code) = diagnostic.code() {
+                    let severity = rules.get_severity_from_code(code.as_str());
+                    if let Some(rule_severity) = severity {
+                        // if the severity of the current diagnostic is different from the severity
+                        // specified in the configuration, we change it
+                        rule_severity
+                    } else {
+                        Severity::Error
+                    }
+                } else {
+                    Severity::Error
+                }
+            } else {
+                Severity::Error
+            };
```

Comment:
This if-else chain could be simplified using combinator methods:
```suggestion
            let severity = rules.as_ref()
                .and_then(|rules| {
                    rules.get_severity_from_code(diagnostic.code()?.as_str())
                })
                .unwrap_or(Severity::Error);
```